src/company_gui.cpp

```diff
@@ -1790,13 +1794,27 @@ struct CompanyInfrastructureWindow : Window
 	RailTypes railtypes; ///< Valid railtypes.
 	RoadTypes roadtypes; ///< Valid roadtypes.
 
-	uint total_width; ///< String width of the total cost line.
+	uint      max_label_width; ///< Infrastructure label size.
+	uint      max_cost_width;  ///< Infratructure maintenance cost text size.
+	uint      max_count_width; ///< Infrastructure count text size.
+
+	Scrollbar *vscroll;        ///< scroll bar widget.
+
+	mutable std::vector<CIWInfrastructureCache> infrastructurelist; ///< infrastructure + total maintenance cost cache vector.
```

Comment:
We try to capitalize comments, and the `+` might be confusing here since it means "and" and not "add". 😉 

```suggestion
	mutable std::vector<CIWInfrastructureCache> infrastructurelist; ///< Infrastructure and total maintenance cost cache vector.
```